docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,557 @@
+`scikit-image` and similar members of the PyData ecosystem. Otherwise, we might arrive to a
+situation where plugin developers are choosing wildly different `numpy` versions for their projects,
+making then non-installable together. In conda jargon, the set of conditional restrains are called
+pinnings and implemented as part of a metapackage (a package that doesn't distribute files, only
+provides metadata). From now on we will refer to them as _napari pinnings_.
+
+
+A prototype notebook assessing the "installability" of all plugins in the same environment
+is available [here](https://colab.research.google.com/drive/1QxbBZYe9-AThGuRsTfwYzT72_UkamXmk)
+Results across Python versions show incompatibilities on Linux, possibly even more intricate on macOS
+and Windows. Excerpt for Python 3.9:
+
+```
+$ mamba create -n test-napari-installability --dry-run -q napari=0.4.15 python=3.9 affinder \
+  bbii-decon brainglobe-napari-io [...] nfinder platelet-unet-watershed smo waver workshop-demo
+Encountered problems while solving.
+Problem: nothing provides __linux needed by dask-cuda-21.10.0-pyhd8ed1ab_0
+Problem: package napari-subboxer-0.0.1-pyhd8ed1ab_0 requires napari 0.4.12, but none of the providers can be installed
+Problem: package napari-tomoslice-0.0.7-pyhd8ed1ab_0 requires napari 0.4.12, but none of the providers can be installed
+Problem: package napari-nikon-nd2-0.1.3-pyhd8ed1ab_0 requires python >=3.6,<=3.9, but none of the providers can be installed
+Problem: package napari-multitask-0.0.2-pyhd8ed1ab_0 requires python >=3.8, but none of the providers can be installed
+Problem: nothing provides __cuda needed by tensorflow-2.7.0-cuda102py310hcf4adbc_0
+Problem: package napari-console-0.0.4-pyhd8ed1ab_0 requires ipykernel >=5.2.0, but none of the providers can be installed
+```
+
+#### Potential Risks
+
+Lack of involvement of the community might mean that only one source of packaging is updated often,
+forcing napari core developers to take over in the maintenance of the conda packages.
+
+Notably, best practices for PyPI packaging are sometimes ignored by some plugins, which means
+that they are not suitable for conda-forge packaging, which has stricter standards for
+publication. The list of problems include vendoring other packages, licensing issues, binary
+redistribution, inadequate dependencies metadata (too strict or too vague). The packaging team
+at napari can help here, but this will not scale if the plugin ecosystem keeps growing.
+
+As a result, some plugins might end up being available on PyPI but not on conda-forge. This further
+reinforces the idea that conda-forge packaging is a second-class citizen for the plugin
+ecosystem. We would recommend making packaging guidelines part of the documentation for plugin
+developers, but also part of the _requirements_ to be accepted on Napari hub listings. Otherwise,
+we risk supporting plugins which do not play nicely with the rest of the ecosystem.
```

Comment:
I'm still pretty opposed to this.  It's one thing to vocally warn end-users than an action they are taking (i.e. installing from pip or a wheel) might break their installation, but it's another thing altogether to not even index things that aren't on conda.  I can even imagine a very small wrapper on our builtin installer that simply errors out if something tries to install over a critical binary (like `numpy`, or `qt`, or any number of things that we observe to have frequent binary incompatibilities).  For the vast majority of pure python plugins, this will _not_ actually be an issue

Reply:
I think we have consensus here that we should not delist, but can have an indicator. Note that this should not prevent merging of the NAP in draft mode — the text can be altered before acceptance and this thread noted by @jaimergp in the discussion/alternatives sections.